src/napari_spatialdata/_view.py

```diff
@@ -256,12 +326,63 @@ def _select_layer(self) -> None:
         else:
             return
 
+    def _update_adata(self) -> None:
+        if (table_name := self.table_name_widget.currentText()) == "":
+            return
+        layer = self._viewer.layers.selection.active
+        adata = None
+
+        if sdata := layer.metadata.get("sdata"):
+            element_name = layer.metadata.get("name")
+            table = sdata[table_name]
+            adata = table[table.obs[table.uns["spatialdata_attrs"]["region_key"]] == element_name]
+            layer.metadata["adata"] = adata
+
+        if layer is not None and "adata" in layer.metadata:
+            with self.model.events.adata.blocker():
+                self.model.adata = layer.metadata["adata"]
+
+        if self.model.adata.shape == (0, 0):
+            return
+
+        self.model.instance_key = layer.metadata["instance_key"] = (
+            adata.uns["spatialdata_attrs"]["instance_key"] if adata is not None else None
+        )
+        self.model.region_key = layer.metadata["region_key"] = (
+            adata.uns["spatialdata_attrs"]["region_key"] if adata is not None else None
+        )
+        self.model.system_name = layer.metadata["name"] if "name" in layer.metadata else None
+
+        if hasattr(
+            self, "obs_widget"
+        ):  # to check if the widget has been already initialized, layer update should only be called on layer change
+            self.adata_layer_widget.clear()
+            self.adata_layer_widget.addItem("X", None)
+            self.adata_layer_widget.addItems(self._get_adata_layer())
+            self.obs_widget._onChange()
+            self.var_widget._onChange()
+            self.obsm_widget._onChange()
+        else:
+            return
+
     def _get_adata_layer(self) -> Sequence[Optional[str]]:
         adata_layers = list(self.model.adata.layers.keys())
         if len(adata_layers):
             return adata_layers
         return [None]
 
+    def _get_init_table_list(self) -> Optional[Sequence[Optional[str]]]:
```

Comment:
also this function is already present above (they are identical up to an empty line present here and not above)